src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOpResolver.scala

```diff
@@ -35,7 +35,7 @@ private[marathon] class InstanceUpdateOpResolver(clock: Clock) extends StrictLog
       case op: Provision =>
         updateExistingInstance(maybeInstance, op.instanceId) { oldInstance =>
           // TODO(karsten): Create events
-          InstanceUpdateEffect.Update(op.instance, oldState = Some(oldInstance), Seq.empty)
+          InstanceUpdateEffect.Update(oldInstance.provisioned(op.agentInfo, op.runSpecVersion, op.tasks, op.now), oldState = Some(oldInstance), Seq.empty)
```

Comment:
Ok, so this is the issue, right? We would override other updates. I think I missed that. However, should we just change two lines. This seems all very overhead. Why not a simple

```scala
val updatedInstance = oldInstance.privisoned(op.agentInfo, op.networkInfo, op.app, op.taskId)
InstanceUpdateEffect.Update(updatedInstance, olderState = Some(oldInstance))
```

Ok, this misses pods but it seems way simpler than all this refactoring.

Reply:
Pushing the tasks is fine for now.